src/components/withKeyboardState.tsx

```diff
@@ -1,20 +1,25 @@
-import React, {forwardRef, createContext, useEffect, useState} from 'react';
+import React, {ComponentType, createContext, ForwardedRef, forwardRef, ReactNode, useEffect, useState} from 'react';
 import {Keyboard} from 'react-native';
 import PropTypes from 'prop-types';
-import getComponentDisplayName from '../libs/getComponentDisplayName';
+import getComponentDisplayName from "../libs/getComponentDisplayName";
 
-const KeyboardStateContext = createContext(null);
-const keyboardStatePropTypes = {
-    /** Whether or not the keyboard is open */
-    isKeyboardShown: PropTypes.bool.isRequired,
+type KeyboardStateContextValue = {
+    /** Whether the keyboard is open */
+    isKeyboardShown : boolean
 };
 
-const keyboardStateProviderPropTypes = {
+type KeyboardStateProviderProps = {
     /* Actual content wrapped by this component */
-    children: PropTypes.node.isRequired,
+    children: ReactNode
 };
 
-function KeyboardStateProvider(props) {
+// TODO: Remove - left for backwards compatibility with existing components.
+const keyboardStatePropTypes = {
+    /** Whether the keyboard is open */
+    isKeyboardShown: PropTypes.bool.isRequired,
+};
+const KeyboardStateContext = createContext<KeyboardStateContextValue | null>(null);
+function KeyboardStateProvider(props: KeyboardStateProviderProps) {
```

Comment:
Add empty lines between different kind of blocks.
```suggestion
const keyboardStatePropTypes = {
    /** Whether the keyboard is open */
    isKeyboardShown: PropTypes.bool.isRequired,
};

const KeyboardStateContext = createContext<KeyboardStateContextValue | null>(null);

function KeyboardStateProvider(props: KeyboardStateProviderProps) {
```